civiform/civiform
# Don't show edit-or-discard-answers modal if applicant hasn't started answering a block

### Description

As part of https://github.com/civiform/civiform/issues/6450, we updated the "Review" and "Previous" buttons to save user answers. If the answers fail validation (e.g. the applicant didn't answer a required question), then we display a modal asking the user to edit their answers or proceed without saving -- see https://github.com/civiform/civiform/blob/main/browser-test/image_snapshots/application_navigation_test/error-on-review-modal.png.

This modal will also pop up if you open a block that you haven't seen before, don't answer any questions, then click "Review" or "Previous". This is pretty unintuitive: applicants may open a block, decide they don't want to fill it out, and click "Review" or "Previous" to see what they've done so far. We should let them proceed to the review page or previous page and not show the modal.

This PR updates the "Review" and "Previous" buttons to **not** show the modal if the applicant opens a block they haven't answered before and fills nothing out. Instead, the applicant is taken directly to the review page / previous page.

Note: This is **not** protected behind the `SAVE_ON_ALL_ACTIONS` feature flag because `if` block I added only triggers when the `ApplicantRequestedAction` is `REVIEW_PAGE` or `PREVIOUS_BLOCK`, and those actions are never used if the `SAVE_ON_ALL_ACTIONS` flag is off.

## Release notes

### Checklist

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)


### Instructions for manual testing

Part 1
- Open a block with required questions, don’t answer any
     - click Review [or Previous] -> verify you go to review page [or previous page] without seeing the modal
     - click Save&Next-> verify you stay on the same page and see errors

Part 2
- Open a block with all optional questions, don’t answer any, click Review [or Previous or Save&next], then fill out other blocks but never reopen that block -> verify you can submit the form

Part 3
- Open a block with required questions, fill them out and save. Re-open that block and delete all answers
     - click Review [or Previous] -> verify error modal appears on top of the block and errors are shown underneath the modal
     - click Save&next ->  verify the errors are immediately shown

Part 4
- Open a block with some optional and some required questions, only answer the required questions,  click Review [or Previous] -> verify you go to review page [or previous page] and your answers were saved. Verify you can submit the form.


### Issue(s) this completes

Fixes #6987
